### DISCUSSION TITLE
Connection problem SUN2000-10KTL-M1

### DISCUSSION THREAD

**Discussion by palecs** [wlcrs/huawei_solar#285] [Setting up the integration] [answered]
Hello, I have SUN2000-10KTL-M1, latest version of donge 133 and latest version of inverter 155. Other modbus tcp tools are working without any issue. Here when I disconnect everything from modbus, restart inverter and try to connect im getting information that cannot connect.

port 502 is opened, tcp modbus working 100%. In log I found that there is is issue as attached. Error are that /usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py on line 288 in _do_read is throwing error and then multiple errors. Thank you for suggestions.

**Comment by palecs** [comment, marked as answer]
Hi solved. slave id is 2, and need to connect 1st time when inverter is producing power. Thank you.

**Comment by zer0coo** [comment]
@palecs do you have any connection issues when you are on Dongle 133?

I loose connection several times per day when im on 133, but if i go back to an older frimware like 127 it works perfect.

I get connection but it is like there is a second connection to Modbus that takes over from time to time when im using 133 firmware.
